src/lib.rs

```diff
@@ -184,6 +179,45 @@ impl FromIterator<AaConfig> for AaSupport {
     }
 }
 
+/// Errors that can occur in vello.
+#[derive(Error, Debug)]
+pub enum VelloError {
+    /// Error when vello couldn't find a wgpu device with suitable features.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't find suitable device")]
+    CouldntFindSuitableDevice,
+    /// Error when wgpu failed to create a surface.
+    /// See [`wgpu::CreateSurfaceError`] for more information.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't create wgpu surface")]
+    WgpuCreateSurfaceError(#[from] wgpu::CreateSurfaceError),
+    /// Error when vello couldn't find [`TextureFormat::Rgba8Unorm`] or [`TextureFormat::Bgra8Unorm`] for your surface.
+    /// Make sure that you have a surface which provides one of those texture formats.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't find `Rgba8Unorm` or `Bgra8Unorm` texture formats for surface")]
+    UnsupportedSurfaceFormat,
+
+    /// Error when using a buffer inside a recording while it's not available.
+    /// Check if you have created it and not freed before its last usage.
+    #[cfg(feature = "wgpu")]
+    #[error("Buffer '{0}' is not available but used for {1}")]
+    UnavailableBufferUsed(&'static str, &'static str),
```

Comment:
Should this only exist if the profiler feature is active, as the docs suggest?

Reply:
Where is that written in the docs?

Reply:
"/// Error when using a buffer inside a recording"

Aren't the recordings related to the `wgpu-profiler` feature? Genuinely not sure, need some guidance.

Reply:
Please don't mark my comments as resolved unless you've responded to them.

Reply:
If it was unclear, I'm asking whether or not this could be `#[cfg(all(feature = "wgpu", feature = "wgpu-profiler"))]`